cpp/tests/utilities_tests/multibuffer_memset_tests.cpp

```diff
@@ -0,0 +1,174 @@
+  auto stream = cudf::get_default_stream();
+  auto _mr    = rmm::mr::get_current_device_resource();
+
+  // Init Host buffers to 0xEE and 0xFF as padding around it
+  std::vector<uint64_t*> host_bufs(NUM_BUFS, nullptr);
+  for (int i = 0; i < NUM_BUFS; i++) {
+    CUDF_CUDA_TRY(cudaMalloc(&host_bufs[i], (BUF_SIZES[i] + 2000) * 8));
+    CUDF_CUDA_TRY(cudaMemsetAsync(host_bufs[i], 0xFF, (BUF_SIZES[i] + 2000) * 8, stream.value()));
+    CUDF_CUDA_TRY(cudaMemsetAsync(host_bufs[i] + 1000, 0xEE, BUF_SIZES[i] * 8, stream.value()));
+  }
+
+  // Init Device buffers
+  std::vector<cudf::device_span<uint64_t>> device_bufs(NUM_BUFS);
+  std::transform(thrust::make_counting_iterator(0L),
+                 thrust::make_counting_iterator(NUM_BUFS),
+                 device_bufs.begin(),
+                 [BUF_SIZES, host_bufs](auto i) {
+                   return cudf::device_span<uint64_t>(host_bufs[i] + 1000, BUF_SIZES[i]);
+                 });
+
+  // Function Call
+  multibuffer_memset(device_bufs, 0UL, stream, _mr);
+
+  // Compare to see that only given buffers are zeroed out
+  std::for_each(thrust::make_counting_iterator(0L),
+                thrust::make_counting_iterator(NUM_BUFS),
+                [BUF_SIZES, host_bufs, device_bufs](auto i) {
+                  std::vector<uint64_t> temp(BUF_SIZES[i] + 2000);
+                  cudf::host_span<uint64_t> host(temp);
+                  CUDF_CUDA_TRY(cudaMemcpy(
+                    host.data(), host_bufs[i], (BUF_SIZES[i] + 2000) * 8, cudaMemcpyDefault));
+                  for (int j = 0; j < BUF_SIZES[i] + 2000; j++) {
+                    if (j < 1000 || j >= BUF_SIZES[i] + 1000) {
+                      EXPECT_EQ(host[j], 0xFFFFFFFFFFFFFFFF);
+                    } else {
+                      EXPECT_EQ(host[j], 0UL);
+                    }
+                  }
+                });
+}
+
+TEST(MultiBufferTestIntegral, BasicTest2)
+{
+  std::vector<long> BUF_SIZES{50000, 4, 1000, 250000, 1, 100, 8000};
+  long NUM_BUFS = BUF_SIZES.size();
+
+  // Device init
+  auto stream = cudf::get_default_stream();
+  auto _mr    = rmm::mr::get_current_device_resource();
+
+  // Init Host buffers to 0xEE and 0xFF as padding around it
+  std::vector<uint64_t*> host_bufs(NUM_BUFS, nullptr);
+  for (int i = 0; i < NUM_BUFS; i++) {
+    CUDF_CUDA_TRY(cudaMalloc(&host_bufs[i], (BUF_SIZES[i] + 2000) * 8));
+    CUDF_CUDA_TRY(cudaMemsetAsync(host_bufs[i], 0xFF, (BUF_SIZES[i] + 2000) * 8, stream.value()));
+    CUDF_CUDA_TRY(cudaMemsetAsync(host_bufs[i] + 1000, 0xEE, BUF_SIZES[i] * 8, stream.value()));
+  }
+
+  // Init Device buffers
+  std::vector<cudf::device_span<uint64_t>> device_bufs(NUM_BUFS);
+  std::transform(thrust::make_counting_iterator(0L),
+                 thrust::make_counting_iterator(NUM_BUFS),
+                 device_bufs.begin(),
+                 [BUF_SIZES, host_bufs](auto i) {
+                   return cudf::device_span<uint64_t>(host_bufs[i] + 1000, BUF_SIZES[i]);
+                 });
```

Comment:
Optional: Looks like a zip iterator may work here.

```suggestion
  std::transform(thrust::make_zip_iterator(thrust::make_tuple(host_bufs.begin(), BUF_SIZES)),
                 thrust::make_zip_iterator(thrust::make_tuple(host_bufs.end(), BUF_SIZES+NUM_BUFS),
                 device_bufs.begin(),
                 [](auto const&elem) {
                   return cudf::device_span<uint64_t>(thrust::get<0>(elem) + 1000, thrust::get<1>(elem));
                 });
```